src/core/file_sys/fs.cpp

```diff
@@ -53,7 +54,13 @@ std::filesystem::path MntPoints::GetHostPath(std::string_view guest_directory, b
     // Remove device (e.g /app0) from path to retrieve relative path.
     pos = mount->mount.size() + 1;
     const auto rel_path = std::string_view(corrected_path).substr(pos);
-    const auto host_path = mount->host_path / rel_path;
+    std::filesystem::path host_path = mount->host_path / rel_path;
+
+    std::filesystem::path patch_path = mount->host_path.string() + "-UPDATE";
+    if (std::filesystem::exists(patch_path / rel_path) && Config::getSeparateUpdateEnabled()) {
+        host_path = patch_path / rel_path;
+    }
+
```

Comment:
It would be better to check if this starts with `/app0/`. The game could also be reading from a save mounted folder

Reply:
Something like "host_path.string().starts_with("/app0/")" in the if?

Reply:
Prob better to use `corrected_path`. Also, host_path.string() can crash when inside a folder with special characters. You could save the parent path and only do the `-UPDATE` concat to the directory name